src/CompositeRepository.php

```diff
@@ -208,10 +238,14 @@ public function listChildren($path)
             ));
         }
 
-        $resources = $this->getRepository($mountPoint)->listChildren($subPath);
-        $this->replaceByReferences($resources, $mountPoint);
+        $collection = $this->getRepository($mountPoint)->listChildren($subPath);
+        $this->replaceByReferences($collection, $mountPoint);
 
-        return $resources;
+        foreach ($collection as $resource) {
+            $compositeCollection->add($resource);
```

Comment:
The mounted resources should probably override any existing resources, this doesn't do that.